ts/IdentificationModal.tsx

```diff
@@ -0,0 +1,191 @@
+
+      const { identificationByPinState } = this.state;
+
+      /**
+       * Create handlers merging default internal actions (to manage the identification state)
+       * with, if available, custom actions passed as props.
+       */
+      const onIdentificationCancelHandler = () => {
+        if (identificationCancelData) {
+          dispatch(identificationCancelData.action);
+        }
+        dispatch(identificationCancel());
+      };
+
+      const onIdentificationSuccessHandler = () => {
+        if (identificationSuccessData) {
+          dispatch(identificationSuccessData.action);
+        }
+        dispatch(identificationSuccess());
+      };
+
+      const onIdentificationFailureHandler = () => {
+        dispatch(identificationFailure());
+      };
+
+      const onPinResetHandler = () => {
+        dispatch(identificationPinReset());
+      };
+
+      return (
+        <Modal
+          visible={true}
+          transparent={true}
+          onRequestClose={onRequestCloseHandler}
+        >
+          <BaseScreenComponent primary={true} contextualHelp={contextualHelp}>
+            <StatusBar barStyle="light-content" />
+            <Content primary={true}>
+              <View spacer={true} extralarge={true} />
+              <Text white={true} alignCenter={true}>
+                {I18n.t("pin_login.pin.pinInfo")}
+              </Text>
+              <Pinpad
+                compareWithCode={pin as string}
+                activeColor={"white"}
+                inactiveColor={"white"}
+                onFulfill={(_: string, __: boolean) =>
+                  this.onPinFullfill(
+                    _,
+                    __,
+                    onIdentificationSuccessHandler,
+                    onIdentificationFailureHandler
+                  )
+                }
+                clearOnInvalid={true}
+              />
+              {renderIdentificationByPinState(identificationByPinState)}
+              <View spacer={true} extralarge={true} />
+              <View>
+                {identificationCancelData !== undefined && (
+                  <Button
+                    block={true}
+                    primary={true}
+                    onPress={onIdentificationCancelHandler}
+                  >
+                    <Text>{identificationCancelData.label}</Text>
+                  </Button>
+                )}
+                {identificationCancelData === undefined && (
+                  <Button
+                    block={true}
+                    primary={true}
+                    onPress={onPinResetHandler}
+                  >
+                    <Text>{I18n.t("pin_login.pin.reset.button")}</Text>
+                  </Button>
+                )}
+                <View spacer={true} />
+                <Text white={true}>{I18n.t("pin_login.pin.reset.tip")}</Text>
+              </View>
+            </Content>
+          </BaseScreenComponent>
+        </Modal>
+      );
+    }
```

Comment:
instead of this huge `if` block, it would be better to negate the predicate and `return null` in the `if`